### PR TITLE
Implement BUILD_LIST_UNPACK

### PR DESCRIPTION
**Pull request by brettcannon**
Part of #106 

If this is accepted then an issue should be opened up about leaking locals (which is only an issue in a OOM situation).

### CODE REVIEW

File: Tests/test_emission.cpp [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        Py_XDECREF(m_jittedcode);
+    }
+
+    std::string& returns() {
+        auto res = run();
+        REQUIRE(res != nullptr);
+        REQUIRE(!PyErr_Occurred());
+
+        auto repr = PyUnicode_AsUTF8(PyObject_Repr(res));
+        Py_DECREF(res);
```
Comment (DinoV) [inline Tests/test_emission.cpp]: I think you need to constructor the std::string before doing the dec ref here, I think the lifetime of the PyUnicode_AsUTF8 is only guaranteed to be valid as long as the object is

File: Tests/testing_util.cpp [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+PyCodeObject* CompileCode(const char* code) {
+    auto globals = PyDict_New();
+    auto builtins = PyThreadState_GET()->interp->builtins;
+    PyDict_SetItemString(globals, "__builtins__", builtins);
+
+    auto locals = PyDict_New();
+    PyRun_String(code, Py_file_input, globals, locals);
+    if (PyErr_Occurred()) {
+        PyErr_Print();
+        Py_DECREF(globals);
```
Comment (DinoV) [inline Tests/testing_util.cpp]: Dec ref locals too

**Comment by DinoV** [comment]
Otherwise LGTM

**Comment by brettcannon** [comment]
I have fixed the refcount issues, but I have discovered that the `float + float` type inference test is now failing with `TypeError: 'int' object is not iterable` which makes no sense since that test has no `int`, let alone something to cause an iteration.

**Comment by brettcannon** [comment]
Turns out the test failure is only when you run all tests; if you run the `[emission]` or `[inference]` tests separately there's no issue. That suggests some problem with state that's carrying over.

**Comment by brettcannon** [comment]
PTAL

**Pull request merged**